Module Readme Revisions (#176)
Azure/azure-saas: src/Saas.Admin/readme.md

3. Saas.Admin.Service.Tests - Unit tests for the service

## 1. Module Overview

> This project hosts a service api which serves as a gateway to administrate the SaaS ecosystem of Tenants. It is fully self-contained such that it includes complete copies of all necessary classes for operation. Since it contains no direct references to the other projects, it can be extracted to launch in isolation insofar as expected substitutes are configured.

Reviewer: ```suggestion
This project hosts a service api which serves as a gateway to administrate the SaaS ecosystem of Tenants. It is fully self-contained such that it includes complete copies of all necessary classes for operation. Since it contains no direct references to the other projects, it can be extracted to launch in isolation. However, keep in mind that some functionality within the API does have [dependencies](https://azure.github.io/azure-saas/components/admin-service/#dependencies) on other services.
```

For a complete overview, please see the [SaaS.Admin.Service](https://azure.github.io/azure-saas/components/admin-service/) page in our documentation site.

## 2. How to Run Locally

> Once configured, this app presents an api service which manages authorization of actions issued to the Admin service. It may be run locally during development of service logic and for regenerating its included NSwag api client. (An NSwag file is included in the Admin project to generate its client.)

Reviewer: This is a bit confusing. Try to stick to a higher level overview.

### i. Requirements

To run the web api, you must have the following installed on your machine:

- [.NET 6.0](https://dotnet.microsoft.com/en-us/download/dotnet/6.0)
- [ASP.NET Core 6.0](https://docs.microsoft.com/en-us/aspnet/core/introduction-to-aspnet-core?view=aspnetcore-6.0)

> - (Reccomended) [Visual Studio](https://visualstudio.microsoft.com/downloads/) or [Visual Studio Code](https://code.visualstudio.com/download)

Reviewer: ```suggestion
- (Recommended) [Visual Studio](https://visualstudio.microsoft.com/downloads/) or [Visual Studio Code](https://code.visualstudio.com/download)
```

- A connection string to a running, empty SQL Server Database.
    - [Local DB](https://docs.microsoft.com/en-us/sql/database-engine/configure-windows/sql-server-express-localdb?view=sql-server-ver15) (Windows Only) - See `Additional Resources` below for basic config secret
    - [SQL Server Docker Container](https://hub.docker.com/_/microsoft-mssql-server)
    - [SQL Server Developer Edition](https://www.microsoft.com/en-us/sql-server/sql-server-downloads)

> - A user store compatible with Microsoft Identity

Reviewer: Change this to be something along the lines of "A deployed instane of the identity framework" and link to the quick start guide on step 1. https://azure.github.io/azure-saas/quick-start/

    - [Azure AD B2C](https://azure.microsoft.com/en-us/services/active-directory/external-identities/b2c/) - created automatically with Bicep deployment

### ii. Development Tools

- [NSwag](https://github.com/RicoSuter/NSwag) - An NSwag configuration file has been included to generate an appropriate client from the included Admin project.
    *Consumes Clients:*

> 	- [permissions-service-client-generator.nswag](Saas.Admin.Service/permissions-service-client-generator.nswag)

Reviewer: I think these files might have gotten misplaced. 

The permissions service client should probably be relocated to live with the permissions service api. 

I think the admin service client was put in the signupadmin site (https://github.com/Azure/azure-saas/blob/main/src/Saas.SignupAdministration/Saas.SignupAdministration.Web/admin-service-client-generator.nswag). This should probably be located in the saas.admin api folder. This is the one you're looking for here, not the permissions service one.

Author: External conversation confirmed this setup was correct, no changes necessary.

| KeyVault:PermissionsApiCert                | The name of the secret in Azure Key Vault that contains a base64 encoded certificate to use for authentication with the permissions api | false  |                                       |
| KeyVault:Url                               | KeyVault URL to pull secret values from in production                                                                                   | false  |                                       |
| Logging:LogLevel:Default                   | Logging level when no configured provider is matched                                                                                    | false  | Information                           |
| Logging:LogLevel:Microsoft.AspNetCore      | Logging level for AspNetCore logging                                                                                                    | false  | Warning                               |
| PermissionsApi:BaseUrl                     | URL for downstream [Permissions API](../Saas.Identity/Saas.Permissions/readme.md)                                                       | false  |                                       |

> | PermissionsApi:LocalCertificate            | A Base64 encoded certificate (.CER) used to authenticate with the permissions API. Only used for local development.                     | true   |                                       |

Reviewer: Add an additional one here, please:

`PermissionsApi:PermissionsApiCertPassphrase`
`The passphrase used to protect the certificate file during creation.`
`secret=true`

> | PermissionsApi:LocalCertificate            | A Base64 encoded certificate (.CER) used to authenticate with the permissions API. Only used for local development.                     | true   |                                       |

Reviewer: ```suggestion
| PermissionsApi:LocalCertificate            | A Base64 encoded certificate (.CER) used to authenticate with the permissions API.                      | true   |                                       |
```
